src/libsync/syncfilestatustracker.cpp

```diff
@@ -236,15 +245,20 @@ void SyncFileStatusTracker::slotAboutToPropagate(SyncFileItemVector &items)
             _syncProblems[item->destination()] = SyncFileStatus::StatusExcluded;
         }
 
+        // Check we are not propagating the same thing twice
+        const auto cleanDestination = destination.endsWith('/') ? destination.chopped(1) : destination;
+        Q_ASSERT(!syncedItemCleanDestinations.contains(cleanDestination));
+        syncedItemCleanDestinations.insert(cleanDestination);
```

Comment:
looking at the code and the comment, we would be doing what the comment says only for debug enabled builds
for the other ones `syncedItemCleanDestinations.insert(cleanDestination);` seems to be overhead
can you check ?

Reply:
my point was that as far as I understand the check is not done for release builds

I think in release builds either the check is done or we do not add entries in `syncedItemCleanDestinations`